### PR TITLE
P5 ap 144 frontend pages challenge details

### PR DESCRIPTION
**Pull request by TimWallaceDev**
JIRA Ticket link: https://makeitmvp.atlassian.net/browse/P5AP-144?atlOrigin=eyJpIjoiNzhlZDViMzg0NDE1NGZjMWI3MzllODEyOWVhYjg4ZDAiLCJwIjoiaiJ9

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Refactoring

## Description

- Changes made? Created a challenge details page
- Reason for changes? User needs a detailed explanation of the challenge 
- Behavior before? There was no details page
- Behavior after? There is a dynamic route that will display the details for any given challenge
- How did you test? Tested manually using development server + web browser

## Checklist

- [ ] I have performed a self-review of my code.
- [x] I have manually tested the new feature and found it to be working as expected.
- [x] All commits in this PR are related to the issue.
- [x] Base branch for this PR is set to `develop`.
- [x] JIRA ticket # and my initials are noted in this PR's name (format example: #57-EB Add Pull Request template)

**Reminder**: every PR needs at least 1 peer approval before merging to the `develop` branch. Once a main ticket has been completed, along with all subtasks, associated code can be merged to `main`.

### CODE REVIEW

File: src/pages/ChallengeDetails/ChallengeDetails.jsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+import bulb from "../../assets/icons/lightbulb.svg";
+import challenges from "./challenges.json";
+
+export function ChallengeDetails() {
+
+
+    const params = useParams()
+    const challengeId = params.challengeId
+
+    // TODO lookup challenge from DB?
```
Comment (robester0403) [inline src/pages/ChallengeDetails/ChallengeDetails.jsx]: Remove comments

File: src/pages/ChallengeDetails/ChallengeDetails.jsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+import DashboardNavbar from "../../components/DashboardNavbar/DashboardNavbar";
+import clock from "../../assets/icons/clock.svg";
+import pen from "../../assets/icons/pen.svg";
+import bulb from "../../assets/icons/lightbulb.svg";
+import challenges from "./challenges.json";
+
+export function ChallengeDetails() {
+
+
+    const params = useParams()
```
Comment (robester0403) [inline src/pages/ChallengeDetails/ChallengeDetails.jsx]: You can destructure the challengeId out.

```suggestion
    const { challengeId } = useParams()
```

File: src/pages/ChallengeDetails/ChallengeDetails.jsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    const challengeId = params.challengeId
+
+    // TODO lookup challenge from DB?
+    // currently just importing mock data array from challenges.json
+
+    const challengeData = challenges.challenges[challengeId]
+    const placeholderData = {
```
Comment (robester0403) [inline src/pages/ChallengeDetails/ChallengeDetails.jsx]: Move this constant out of this file

File: src/pages/ChallengeDetails/ChallengeDetails.jsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+                                <span className="overview__type-text">VSCode / Figma</span>
+                            </div>
+                        </div>
+
+                        <div className="objectives">
+                            <h2 className="objectives__title">Constraints</h2>
+                            <ul className="objectives__list">
+                                {challengeData.detailedProblem.constraints.map(constraint => {
+                                    return (
+                                        <li className="objectives__item" key={constraint}>{constraint}</li>
```
Comment (robester0403) [inline src/pages/ChallengeDetails/ChallengeDetails.jsx]: constraint should not be used as a key. at worst use index as the key

File: src/pages/ChallengeDetails/ChallengeDetails.jsx [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+                                })}
+                            </ul>
+                        </div>
+
+                        <div className="examples">
+                            <h2 className="examples__title">Examples</h2>
+                            <ul className="examples__list">
+                                {challengeData.detailedProblem.examples.map(example => {
+                                    return (
+                                        <code className="examples__item" key={example.input}>
```
Comment (robester0403) [inline src/pages/ChallengeDetails/ChallengeDetails.jsx]: Key should not be input ideally

File: src/pages/ChallengeDetails/ChallengeDetails.jsx [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+                                })}
+                            </ul>
+                        </div>
+
+                        <div className="solutions">
+                            <h2 className="solutions__title">Solution Approach</h2>
+                            <ul className="solutions__list">
+                                {challengeData.detailedProblem.solutionApproach.map(example => {
+                                    return (
+                                        <li className="solutions__item" key={example}>
```
Comment (robester0403) [inline src/pages/ChallengeDetails/ChallengeDetails.jsx]: You should not use this as key

File: src/pages/ChallengeDetails/ChallengeDetails.jsx (hunk 1 above)
Comment (TimWallaceDev) [reply src/pages/ChallengeDetails/ChallengeDetails.jsx]: Comments are now removed

File: src/pages/ChallengeDetails/ChallengeDetails.jsx (hunk 2 above)
Comment (TimWallaceDev) [reply src/pages/ChallengeDetails/ChallengeDetails.jsx]: ChallengeId is now desctructured

File: src/pages/ChallengeDetails/ChallengeDetails.jsx (hunk 3 above)
Comment (TimWallaceDev) [reply src/pages/ChallengeDetails/ChallengeDetails.jsx]: Done

File: src/pages/ChallengeDetails/ChallengeDetails.jsx (hunk 4 above)
Comment (TimWallaceDev) [reply src/pages/ChallengeDetails/ChallengeDetails.jsx]: I changed the key to the index. Would it be ideal to generate a UID for each constraint?

File: src/pages/ChallengeDetails/ChallengeDetails.jsx (hunk 5 above)
Comment (TimWallaceDev) [reply src/pages/ChallengeDetails/ChallengeDetails.jsx]: Key is changed to the index now

File: src/pages/ChallengeDetails/ChallengeDetails.jsx (hunk 6 above)
Comment (TimWallaceDev) [reply src/pages/ChallengeDetails/ChallengeDetails.jsx]: Key is changed to the index now

**Review by robester0403 — APPROVED** [review]

**Pull request merged**